tests/console/validate_encrypt.pm

```diff
@@ -19,30 +19,72 @@ use strict;
 use warnings;
 use base "opensusebasetest";
 use testapi;
+use Test::Assert ':all';
+use scheduler 'get_test_data';
 
-sub run {
+sub parse_crypttab {
+    my $path = shift;
+    return {split(/\s+/, script_output("cat $path"))};
```

Comment:
I'm not sure if this parsing scales.
Consider following example:
```perl
my $message = <<'END_MESSAGE';
luks       UUID=2505567a-9e27-4efe-a4d5-15ad146c258b
swap       /dev/sda7       /dev/urandom       swap
truecrypt  /dev/sda2       /etc/container_password  tcrypt
hidden     /mnt/tc_hidden  /dev/null    tcrypt-hidden,tcrypt-keyfile=/etc/keyfile
external   /dev/sda3       keyfile:LABEL=keydev keyfile-timeout=10s
END_MESSAGE

my $hash = {split(/\s+/, $message)};

print Dumper($hash);
```
I won't get a single disk as a key:
```
$VAR1 = {
          '/dev/urandom' => 'swap',
          'external' => '/dev/sda3',
          '/etc/container_password' => 'tcrypt',
          'truecrypt' => '/dev/sda2',
          'keyfile:LABEL=keydev' => 'keyfile-timeout=10s',
          'hidden' => '/mnt/tc_hidden',
          'luks' => 'UUID=2505567a-9e27-4efe-a4d5-15ad146c258b',
          '/dev/null' => 'tcrypt-hidden,tcrypt-keyfile=/etc/keyfile',
          'swap' => '/dev/sda7'
        };
```
So we might check if device is mentioned in the config file or do more complex parsing.

Reply:
I should have looked at the documentation, good catch! I misnamed my idea of 'test scaling', I meant more like we should have stuff in one method separately to parse the command line and then spend some time in the regex, sometimes is ok, sometimes like in this case, not enough if something want to try to encrypt swap or something more difficult. But I see that with this structure you were able to use the Dumper easly to test that it was not working what I think it was also my point that could make easy debug this test.
Now should be more than fine the parsing! :) (I hope) I checked with this:
```
#!/usr/bin/perl
 
use Data::Dumper;
 
my $message = <<'END_MESSAGE';
 
#luks       UUID=111111111-4e111fe-a411d5-1511adc258b
luks       UUID=2505567a-9e27-4efe-a4d5-15ad146c258b
swap       /dev/sda7       /dev/urandom       swap
truecrypt  /dev/sda2       /etc/container_password  tcrypt
 # truecrypt  /dev/sda7       /etc/container_login  tcrypt
hidden     /mnt/tc_hidden  /dev/null    tcrypt-hidden,tcrypt-keyfile=/etc/keyfile
external   /dev/sda3       keyfile:LABEL=keydev keyfile-timeout=10s
 
external2  /dev/sda4       keyfile:LABEL=keydev2
END_MESSAGE
 
my @lines = split(/\n/, $message);
my @crypttab = ();
foreach(@lines) {
  next if /^\s*#.*$/;
  if($_ =~ /(?<name>.+?)\s+(?<encrypted_device>.+?)($|\s+(?<password>.*?)($|\s+(?<options>.*)))/) {
    my ($name, $encrypted_device) = ($+{name}, $+{encrypted_device});
    my ($password, $options) = ($+{password}, $+{options});
    push @crypttab,
         { name => $name, encrypted_device => $encrypted_device,
           password => $password, options => $options
         };
  }
}
print Dumper(\@crypttab);
```